python3Packages.sip: 5.5.0 -> 6.1.1, pythonPackages.pyqt5: 5.15.2 -> 5.15.4, python3Packages.pyqt-builder: 1.6.0 -> 1.10.1

eduardosm opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

eduardosm committed:
pythonPackages.pyqt5: 5.15.2 -> 5.15.4

eduardosm committed:
python3Packages.sip: 5.5.0 -> 6.1.1

eduardosm committed:
python3Packages.pyqt5: build using `pyproject`

Required to use sip 6

eduardosm committed:
python3Packages.pyqt-builder: 1.6.0 -> 1.10.1

eduardosm committed:
python3Packages.qscintilla-qt5: fix PyQt5 SIP path

eduardosm committed:
python3Packages.pyqtwebengine: build using `pyproject`

eduardosm committed:
calibre: fix PyQt5 SIP path

eduardosm committed:
krita: fix PyQt5 SIP path

eduardosm committed:
qgis: fix PyQt5 SIP path

eduardosm committed:
pythonPackages.pyqtwebengine: disable python 2 support

Nothing depends on it and simplifies the definition significantly

eduardosm commented:
Result of `nixpkgs-review pr 123378` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>multibootusb</li>
    <li>nagstamon</li>
    <li>python38Packages.skein</li>
    <li>python39Packages.skein</li>
    <li>tribler</li>
    <li>webbrowser</li>
  </ul>
</details>
<details>
  <summary>165 packages built:</summary>
  <ul>
    <li>aj-snapshot</li>
    <li>anki</li>
    <li>ankisyncd</li>
    <li>asymptote</li>
    <li>audiality2</li>
    <li>backintime</li>
    <li>cadence</li>
    <li>calibre</li>
    <li>carla</li>
    <li>convertall</li>
    <li>corrscope</li>
    <li>cq-editor</li>
    <li>cura</li>
    <li>curaLulzbot</li>
    <li>dupeguru</li>
    <li>eddy</li>
    <li>electron-cash</li>
    <li>electrum</li>
    <li>electrum-ltc</li>
    <li>faust2jack</li>
    <li>faust2jaqt</li>
    <li>ffado (ffadoFull)</li>
    <li>flent</li>
    <li>frescobaldi</li>
    <li>friture</li>
    <li>giada</li>
    <li>git-annex-metadata-gui</li>
    <li>git-cola</li>
    <li>gmic-qt-krita (gmic_krita_qt)</li>
    <li>gns3-gui</li>
    <li>gnss-sdr</li>
    <li>gnuradio</li>
    <li>gnuradio3_8</li>
    <li>gnuradio3_8Packages.gnuradio</li>
    <li>gnuradio3_8Packages.limesdr</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gnuradio3_8Packages.rds</li>
    <li>gnuradioPackages.gnuradio</li>
    <li>hplip</li>
    <li>hplipWithPlugin</li>
    <li>ihaskell</li>
    <li>inkcut</li>
    <li>inkscape-extensions.inkcut</li>
    <li>jack1</li>
    <li>jack2 (jack2Full)</li>
    <li>jackmix_jack1</li>
    <li>jellyfin-mpv-shim</li>
    <li>kcc</li>
    <li>kmymoney</li>
    <li>koboredux-free</li>
    <li>krita</li>
    <li>krop</li>
    <li>labelImg</li>
    <li>leo-editor</li>
    <li>littlegptracker</li>
    <li>lsp-plugins</li>
    <li>ltc-tools</li>
    <li>luppp</li>
    <li>maestral-gui</li>
    <li>magnetophonDSP.CharacterCompressor</li>
    <li>magnetophonDSP.CompBus</li>
    <li>magnetophonDSP.ConstantDetuneChorus</li>
    <li>magnetophonDSP.LazyLimiter</li>
    <li>magnetophonDSP.MBdistortion</li>
    <li>magnetophonDSP.RhythmDelay</li>
    <li>magnetophonDSP.VoiceOfFaust</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>magnetophonDSP.pluginUtils</li>
    <li>magnetophonDSP.shelfMultiBand</li>
    <li>manuskript</li>
    <li>mkchromecast</li>
    <li>mnemosyne</li>
    <li>mooSpace</li>
    <li>nanovna-saver</li>
    <li>octavePackages.audio</li>
    <li>onionshare-gui</li>
    <li>openlp</li>
    <li>openlpFull</li>
    <li>openshot-qt</li>
    <li>opensnitch-ui</li>
    <li>persepolis</li>
    <li>picard</li>
    <li>plover.dev</li>
    <li>puddletag</li>
    <li>pulseeffects-legacy</li>
    <li>pulseeffects-pw</li>
    <li>pyspread</li>
    <li>python38Packages.ansible-kernel</li>
    <li>python38Packages.binwalk-full</li>
    <li>python38Packages.enamlx</li>
    <li>python38Packages.jupyter</li>
    <li>python38Packages.lightparam</li>
    <li>python38Packages.mayavi</li>
    <li>python38Packages.poppler-qt5</li>
    <li>python38Packages.pyqt-builder</li>
    <li>python38Packages.pyqt5</li>
    <li>python38Packages.pyqt5_with_qtmultimedia</li>
    <li>python38Packages.pyqt5_with_qtwebkit</li>
    <li>python38Packages.pyqtgraph</li>
    <li>python38Packages.pyqtwebengine</li>
    <li>python38Packages.pywebview</li>
    <li>python38Packages.qimage2ndarray</li>
    <li>python38Packages.qreactor</li>
    <li>python38Packages.qscintilla-qt5</li>
    <li>python38Packages.qtconsole</li>
    <li>python38Packages.quamash</li>
    <li>python38Packages.sip</li>
    <li>spyder (python38Packages.spyder)</li>
    <li>python38Packages.spyder_3</li>
    <li>python38Packages.stytra</li>
    <li>python38Packages.uranium</li>
    <li>python38Packages.vega</li>
    <li>python38Packages.weboob</li>
    <li>python38Packages.woob</li>
    <li>python39Packages.ansible-kernel</li>
    <li>python39Packages.binwalk-full</li>
    <li>python39Packages.enamlx</li>
    <li>python39Packages.jupyter</li>
    <li>python39Packages.lightparam</li>
    <li>python39Packages.mayavi</li>
    <li>python39Packages.poppler-qt5</li>
    <li>python39Packages.pyqt-builder</li>
    <li>python39Packages.pyqt5</li>
    <li>python39Packages.pyqt5_with_qtmultimedia</li>
    <li>python39Packages.pyqt5_with_qtwebkit</li>
    <li>python39Packages.pyqtgraph</li>
    <li>python39Packages.pyqtwebengine</li>
    <li>python39Packages.pywebview</li>
    <li>python39Packages.qimage2ndarray</li>
    <li>python39Packages.qreactor</li>
    <li>python39Packages.qscintilla-qt5</li>
    <li>python39Packages.qtconsole</li>
    <li>python39Packages.quamash</li>
    <li>python39Packages.sip</li>
    <li>python39Packages.spyder_3</li>
    <li>python39Packages.uranium</li>
    <li>python39Packages.vega</li>
    <li>python39Packages.weboob</li>
    <li>python39Packages.woob</li>
    <li>qarte</li>
    <li>qgis</li>
    <li>qgis-unwrapped</li>
    <li>qnotero</li>
    <li>qpaeq</li>
    <li>qradiolink</li>
    <li>qutebrowser</li>
    <li>rapid-photo-downloader</li>
    <li>retext</li>
    <li>rmview</li>
    <li>scudcloud</li>
    <li>shotcut</li>
    <li>sonic-pi</li>
    <li>soundtracker</li>
    <li>sumorobot-manager</li>
    <li>tambura</li>
    <li>tetraproc</li>
    <li>tiscamera</li>
    <li>tortoisehg</li>
    <li>tuxguitar</li>
    <li>urh</li>
    <li>vimiv-qt</li>
    <li>virtscreen</li>
    <li>vorta</li>
    <li>webcamoid</li>
    <li>webmacs</li>
  </ul>
</details>

r-rmcgibbo commented:
Result of `nixpkgs-review pr 123378` at 48582996 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages marked as broken and skipped:</summary>
  <ul>
    <li>asymptote</li>
    <li>giada</li>
    <li>hplipWithPlugin_3_18_5</li>
    <li>hplip_3_18_5</li>
    <li>multibootusb</li>
    <li>nagstamon</li>
    <li>pulseeffects-legacy</li>
    <li>pulseeffects-pw</li>
    <li>python38Packages.skein</li>
    <li>python39Packages.skein</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/975c28db75bdd1dbadd7adc95eead4b4">onionshare-gui</a></li>
  </ul>
</details>
<details>
  <summary>58 packages skipped due to time constraints:</summary>
  <ul>
    <li>carla</li>
    <li>cq-editor</li>
    <li>electrum</li>
    <li>electrum-ltc</li>
    <li>gmic-qt-krita</li>
    <li>gns3-gui</li>
    <li>gnss-sdr</li>
    <li>gnuradio</li>
    <li>gnuradio3_8</li>
    <li>gnuradio3_8Packages.gnuradio</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>82 packages built successfully:</summary>
  <ul>
    <li>aj-snapshot</li>
    <li>anki</li>
    <li>ankisyncd</li>
    <li>audiality2</li>
    <li>backintime (backintime-qt)</li>
    <li>calibre</li>
    <li>convertall</li>
    <li>corrscope</li>
    <li>cura</li>
    <li>curaLulzbot</li>
    <li>dupeguru</li>
    <li>eddy</li>
    <li>electron-cash</li>
    <li>faust2jack</li>
    <li>faust2jaqt</li>
    <li>ffado (libffado)</li>
    <li>flent</li>
    <li>frescobaldi</li>
    <li>friture</li>
    <li>git-annex-metadata-gui</li>
    <li>git-cola</li>
    <li>hplip</li>
    <li>inkcut</li>
    <li>jack1</li>
    <li>jack2</li>
    <li>jackmix_jack1</li>
    <li>kcc</li>
    <li>krop</li>
    <li>labelImg</li>
    <li>littlegptracker</li>
    <li>ltc-tools</li>
    <li>luppp</li>
    <li>maestral-gui</li>
    <li>manuskript</li>
    <li>mkchromecast</li>
    <li>mnemosyne</li>
    <li>nanovna-saver</li>
    <li>octavePackages.audio</li>
    <li>opensnitch-ui</li>
    <li>persepolis</li>
    <li>puddletag</li>
    <li>pyspread</li>
    <li>python38Packages.ansible-kernel</li>
    <li>python38Packages.binwalk-full</li>
    <li>python38Packages.enamlx</li>
    <li>python38Packages.jupyter</li>
    <li>python38Packages.lightparam</li>
    <li>python38Packages.mayavi</li>
    <li>python38Packages.poppler-qt5</li>
    <li>python38Packages.pyqt-builder</li>
    <li>python38Packages.pyqt5</li>
    <li>python38Packages.pyqtgraph</li>
    <li>python38Packages.pyqtwebengine</li>
    <li>python38Packages.qimage2ndarray</li>
    <li>python38Packages.qreactor</li>
    <li>python38Packages.qscintilla-qt5</li>
    <li>python38Packages.qtconsole</li>
    <li>python38Packages.quamash</li>
    <li>python38Packages.sip</li>
    <li>spyder (python38Packages.spyder)</li>
    <li>python38Packages.spyder_3</li>
    <li>python38Packages.stytra</li>
    <li>python38Packages.uranium</li>
    <li>python38Packages.vega</li>
    <li>python38Packages.weboob</li>
    <li>python38Packages.woob</li>
    <li>python39Packages.pyqt-builder</li>
    <li>python39Packages.sip</li>
    <li>qnotero</li>
    <li>qpaeq</li>
    <li>qutebrowser</li>
    <li>rapid-photo-downloader</li>
    <li>retext</li>
    <li>rmview</li>
    <li>soundtracker</li>
    <li>tetraproc</li>
    <li>tortoisehg</li>
    <li>urh</li>
    <li>vimiv-qt</li>
    <li>virtscreen</li>
    <li>vorta</li>
    <li>webmacs</li>
  </ul>
</details>
<details>
  <summary>5 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/pyqt-builder/default.nix:20:3:
```
   |
20 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/qscintilla-qt5/default.nix:52:0:
```
   |
52 |     description = "A Python binding to QScintilla, Qt based text editing control";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/pyqtwebengine/default.nix:64:3:
```
   |
64 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/pyqt/5.x.nix:162:5:
```
    |
162 |     license     = licenses.gpl3;
    |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/pyqtwebengine/default.nix:67:5:
```
   |
67 |     license     = licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

eduardosm committed:
pythonPackages.pyqt-builder: add eduardosm as maintainer

eduardosm committed:
pythonPackages.qscintilla-qt5: add pythonImportsCheck and disable Python 2 support

eduardosm committed:
pythonPackages.pyqt5: fix license and homepage

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/539

SuperSandro2000 commented on pkgs/development/python-modules/pyqt/5.x.nix:
```diff
@@ -155,7 +159,7 @@ in buildPythonPackage rec {
   meta = with lib; {
     description = "Python bindings for Qt5";
     homepage    = "http://www.riverbankcomputing.co.uk";
```
```suggestion
    homepage    = "https://riverbankcomputing.com/";
```

eduardosm replied:
Done, also fixed for `pythonPackages.sip`

eduardosm committed:
pythonPackages.sip{,_4}: update homepage

dotlambda commented:
Do we still want to support Python 2? It would simplify our expression for PyQt5 a lot if we don't.

eduardosm commented:
> Do we still want to support Python 2? It would simplify our expression for PyQt5 a lot if we don't.

Indeed it would. There is an [old hplip version](https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/drivers/hplip/3.18.5.nix) that uses python2-pyqt5, but I don't know the reason for keeping that version and whether it is maintained.

dotlambda commented:
> There is an [old hplip version](https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/drivers/hplip/3.18.5.nix) that uses python2-pyqt5, but I don't know the reason for keeping that version and whether it is maintained.

It doesn't seem to be used. So let's go ahead and remove Python 2 support from PyQt5 :tada:

eduardosm committed:
hplip_3_18_5: remove

It does not seem to be used anymore and will allow to simplify pyqt5's expression (https://github.com/NixOS/nixpkgs/pull/123378#issuecomment-866786136).

eduardosm committed:
pythonPackages.pyqt5: remove Python 2 support

Not used and simplifies the expression.

eduardosm commented:
> > There is an [old hplip version](https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/drivers/hplip/3.18.5.nix) that uses python2-pyqt5, but I don't know the reason for keeping that version and whether it is maintained.
> 
> It doesn't seem to be used. So let's go ahead and remove Python 2 support from PyQt5 tada

Done

SuperSandro2000 commented on pkgs/development/python-modules/qscintilla-qt5/default.nix:
```diff
@@ -4,12 +4,14 @@
 , qtbase
 }:
 with pythonPackages;
```
I am pretty sure this is not allowed in pythonPackages.

eduardosm replied:
I do not like it neither, fixed.

eduardosm committed:
pythonPackages.qscintilla-qt5: do not use `with pythonPackages;`

SuperSandro2000 commented on pkgs/development/python-modules/pyqt/5.x.nix:
```diff
@@ -13,31 +13,15 @@
-        EOF
-      '';
-    });
+  inherit (pythonPackages) buildPythonPackage python isPy3k dbus-python sip pyqt-builder;
```
The problem is that this package is using libsForQT5.callPackage https://github.com/NixOS/nixpkgs/blob/1c203a232544018680b757443083262a1bff8f20/pkgs/top-level/python-packages.nix#L6332

I thought it used the python callPackages.

eduardosm replied:
Would it be better if it used python callPackages?

SuperSandro2000 replied:
There is probably a reason why it is like this and I think it is to support multiple qt5 versions. If there is not a good reason it would be good if it would be converted.

eduardosm replied:
That way it only supports whatever version `libsForQt5` is, so I changed it to use python callPackage.

eduardosm committed:
veusz: fix PyQt5 sip path and use sip 4

veusz tries to use the `sip` binary, which is no longer available in sip 6

eduardosm committed:
python3Packages.pyqt5: use python callPackage instead of qt5 callPackage

eduardosm commented:
Result of `nixpkgs-review pr 123378` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>multibootusb</li>
    <li>nagstamon</li>
    <li>python38Packages.skein</li>
    <li>python39Packages.skein</li>
    <li>tribler</li>
    <li>webbrowser</li>
  </ul>
</details>
<details>
  <summary>171 packages built:</summary>
  <ul>
    <li>aj-snapshot</li>
    <li>anki</li>
    <li>ankisyncd</li>
    <li>asymptote</li>
    <li>audiality2</li>
    <li>backintime</li>
    <li>cadence</li>
    <li>calibre</li>
    <li>carla</li>
    <li>convertall</li>
    <li>corrscope</li>
    <li>cq-editor</li>
    <li>cura</li>
    <li>curaLulzbot</li>
    <li>dupeguru</li>
    <li>eddy</li>
    <li>electron-cash</li>
    <li>electrum</li>
    <li>electrum-ltc</li>
    <li>faust2jack</li>
    <li>faust2jaqt</li>
    <li>faustPhysicalModeling</li>
    <li>faustStk</li>
    <li>ffado (ffadoFull)</li>
    <li>flent</li>
    <li>flowblade</li>
    <li>frescobaldi</li>
    <li>friture</li>
    <li>giada</li>
    <li>git-annex-metadata-gui</li>
    <li>git-cola</li>
    <li>gmic-qt-krita (gmic_krita_qt)</li>
    <li>gns3-gui</li>
    <li>gnss-sdr</li>
    <li>gnuradio</li>
    <li>gnuradio3_8</li>
    <li>gnuradio3_8Packages.ais</li>
    <li>gnuradio3_8Packages.gnuradio</li>
    <li>gnuradio3_8Packages.limesdr</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gnuradio3_8Packages.rds</li>
    <li>gnuradioPackages.gnuradio</li>
    <li>hplip</li>
    <li>hplipWithPlugin</li>
    <li>ihaskell</li>
    <li>inkcut</li>
    <li>inkscape-extensions.inkcut</li>
    <li>jack1</li>
    <li>jack2 (jack2Full)</li>
    <li>jackmix_jack1</li>
    <li>jellyfin-mpv-shim</li>
    <li>kcc</li>
    <li>kmymoney</li>
    <li>koboredux-free</li>
    <li>krita</li>
    <li>krop</li>
    <li>labelImg</li>
    <li>leo-editor</li>
    <li>littlegptracker</li>
    <li>lsp-plugins</li>
    <li>ltc-tools</li>
    <li>luppp</li>
    <li>maestral-gui</li>
    <li>magnetophonDSP.CharacterCompressor</li>
    <li>magnetophonDSP.CompBus</li>
    <li>magnetophonDSP.ConstantDetuneChorus</li>
    <li>magnetophonDSP.LazyLimiter</li>
    <li>magnetophonDSP.MBdistortion</li>
    <li>magnetophonDSP.RhythmDelay</li>
    <li>magnetophonDSP.VoiceOfFaust</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>magnetophonDSP.pluginUtils</li>
    <li>magnetophonDSP.shelfMultiBand</li>
    <li>manuskript</li>
    <li>mkchromecast</li>
    <li>mlt</li>
    <li>mnemosyne</li>
    <li>mooSpace</li>
    <li>nanovna-saver</li>
    <li>octavePackages.audio</li>
    <li>onionshare-gui</li>
    <li>openlp</li>
    <li>openlpFull</li>
    <li>openshot-qt</li>
    <li>opensnitch-ui</li>
    <li>persepolis</li>
    <li>picard</li>
    <li>plover.dev</li>
    <li>puddletag</li>
    <li>pulseeffects-legacy</li>
    <li>pulseeffects-pw</li>
    <li>pyspread</li>
    <li>python38Packages.ansible-kernel</li>
    <li>python38Packages.binwalk-full</li>
    <li>python38Packages.enamlx</li>
    <li>python38Packages.jupyter</li>
    <li>python38Packages.lightparam</li>
    <li>python38Packages.mayavi</li>
    <li>python38Packages.mlt</li>
    <li>python38Packages.poppler-qt5</li>
    <li>python38Packages.pyqt-builder</li>
    <li>python38Packages.pyqt5</li>
    <li>python38Packages.pyqt5_with_qtmultimedia</li>
    <li>python38Packages.pyqt5_with_qtwebkit</li>
    <li>python38Packages.pyqtgraph</li>
    <li>python38Packages.pyqtwebengine</li>
    <li>python38Packages.pywebview</li>
    <li>python38Packages.qimage2ndarray</li>
    <li>python38Packages.qreactor</li>
    <li>python38Packages.qscintilla-qt5</li>
    <li>python38Packages.qtconsole</li>
    <li>python38Packages.quamash</li>
    <li>python38Packages.sip</li>
    <li>spyder (python38Packages.spyder)</li>
    <li>python38Packages.stytra</li>
    <li>python38Packages.uranium</li>
    <li>python38Packages.vega</li>
    <li>python38Packages.weboob</li>
    <li>python38Packages.woob</li>
    <li>python39Packages.ansible-kernel</li>
    <li>python39Packages.binwalk-full</li>
    <li>python39Packages.enamlx</li>
    <li>python39Packages.jupyter</li>
    <li>python39Packages.lightparam</li>
    <li>python39Packages.mayavi</li>
    <li>python39Packages.mlt</li>
    <li>python39Packages.poppler-qt5</li>
    <li>python39Packages.pyqt-builder</li>
    <li>python39Packages.pyqt5</li>
    <li>python39Packages.pyqt5_with_qtmultimedia</li>
    <li>python39Packages.pyqt5_with_qtwebkit</li>
    <li>python39Packages.pyqtgraph</li>
    <li>python39Packages.pyqtwebengine</li>
    <li>python39Packages.pywebview</li>
    <li>python39Packages.qimage2ndarray</li>
    <li>python39Packages.qreactor</li>
    <li>python39Packages.qscintilla-qt5</li>
    <li>python39Packages.qtconsole</li>
    <li>python39Packages.quamash</li>
    <li>python39Packages.sip</li>
    <li>python39Packages.uranium</li>
    <li>python39Packages.vega</li>
    <li>python39Packages.weboob</li>
    <li>python39Packages.woob</li>
    <li>qarte</li>
    <li>qgis</li>
    <li>qgis-unwrapped</li>
    <li>qnotero</li>
    <li>qpaeq</li>
    <li>qradiolink</li>
    <li>qutebrowser</li>
    <li>rapid-photo-downloader</li>
    <li>retext</li>
    <li>rmview</li>
    <li>scudcloud</li>
    <li>shotcut</li>
    <li>sonic-pi</li>
    <li>soundtracker</li>
    <li>sumorobot-manager</li>
    <li>tambura</li>
    <li>tetraproc</li>
    <li>tiscamera</li>
    <li>tortoisehg</li>
    <li>tuxguitar</li>
    <li>urh</li>
    <li>veusz</li>
    <li>vimiv-qt</li>
    <li>virtscreen</li>
    <li>vorta</li>
    <li>webcamoid</li>
    <li>webmacs</li>
  </ul>
</details>

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 123378` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages marked as broken and skipped:
</summary>
<ul>
<li>multibootusb</li>
<li>nagstamon</li>
<li>python38Packages.skein</li>
<li>python39Packages.skein</li>
<li>tribler</li>
<li>webbrowser</li>
</ul>
</details>
<details>

  <summary>7 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>corrscope: log was empty</li>
<li>cq-editor: log was empty</li>
<li>electron-cash: log was empty</li>
<li>electrum: log was empty</li>
<li>electrum-ltc: log was empty</li>
<li>onionshare-gui: log was empty</li>
<li>spyder: log was empty</li>
  </ul>
</details>
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>koboredux: log was empty</li>
<li>(python38Packages.spyder): log was empty</li>
  </ul>
</details>
<details>
<summary>
162 packages built:
</summary>
<ul>
<li>aj-snapshot</li>
<li>anki</li>
<li>ankisyncd</li>
<li>asymptote</li>
<li>audiality2</li>
<li>backintime (backintime-qt)</li>
<li>cadence</li>
<li>calibre</li>
<li>carla</li>
<li>convertall</li>
<li>cura</li>
<li>curaLulzbot</li>
<li>dupeguru</li>
<li>eddy</li>
<li>faust2jack</li>
<li>faust2jaqt</li>
<li>faustPhysicalModeling</li>
<li>faustStk</li>
<li>ffado (libffado)</li>
<li>flent</li>
<li>flowblade</li>
<li>frescobaldi</li>
<li>friture</li>
<li>giada</li>
<li>git-annex-metadata-gui</li>
<li>git-cola</li>
<li>gmic-qt-krita</li>
<li>gns3-gui</li>
<li>gnss-sdr</li>
<li>gnuradio</li>
<li>gnuradio3_8</li>
<li>gnuradio3_8Packages.ais</li>
<li>gnuradio3_8Packages.gnuradio</li>
<li>gnuradio3_8Packages.limesdr</li>
<li>gnuradio3_8Packages.osmosdr</li>
<li>gnuradio3_8Packages.rds</li>
<li>gnuradioPackages.gnuradio</li>
<li>hplip</li>
<li>hplipWithPlugin</li>
<li>ihaskell</li>
<li>inkcut</li>
<li>inkscape-extensions.inkcut</li>
<li>jack1</li>
<li>jack2</li>
<li>jackmix_jack1</li>
<li>jellyfin-mpv-shim</li>
<li>kcc</li>
<li>kmymoney</li>
<li>koboredux-free</li>
<li>krita</li>
<li>krop</li>
<li>labelImg</li>
<li>leo-editor</li>
<li>littlegptracker</li>
<li>lsp-plugins</li>
<li>ltc-tools</li>
<li>luppp</li>
<li>maestral-gui</li>
<li>magnetophonDSP.CharacterCompressor</li>
<li>magnetophonDSP.CompBus</li>
<li>magnetophonDSP.ConstantDetuneChorus</li>
<li>magnetophonDSP.LazyLimiter</li>
<li>magnetophonDSP.MBdistortion</li>
<li>magnetophonDSP.RhythmDelay</li>
<li>magnetophonDSP.VoiceOfFaust</li>
<li>magnetophonDSP.faustCompressors</li>
<li>magnetophonDSP.pluginUtils</li>
<li>magnetophonDSP.shelfMultiBand</li>
<li>manuskript</li>
<li>mkchromecast</li>
<li>mlt</li>
<li>mnemosyne</li>
<li>mooSpace</li>
<li>nanovna-saver</li>
<li>octavePackages.audio</li>
<li>openshot-qt</li>
<li>opensnitch-ui</li>
<li>persepolis</li>
<li>picard</li>
<li>plover.dev</li>
<li>puddletag</li>
<li>pulseeffects-legacy</li>
<li>pulseeffects-pw</li>
<li>pyspread</li>
<li>python38Packages.ansible-kernel</li>
<li>python38Packages.binwalk-full</li>
<li>python38Packages.enamlx</li>
<li>python38Packages.jupyter</li>
<li>python38Packages.lightparam</li>
<li>python38Packages.mayavi</li>
<li>python38Packages.mlt</li>
<li>python38Packages.poppler-qt5</li>
<li>python38Packages.pyqt-builder</li>
<li>python38Packages.pyqt5</li>
<li>python38Packages.pyqt5_with_qtmultimedia</li>
<li>python38Packages.pyqt5_with_qtwebkit</li>
<li>python38Packages.pyqtgraph</li>
<li>python38Packages.pyqtwebengine</li>
<li>python38Packages.pywebview</li>
<li>python38Packages.qimage2ndarray</li>
<li>python38Packages.qreactor</li>
<li>python38Packages.qscintilla-qt5</li>
<li>python38Packages.qtconsole</li>
<li>python38Packages.quamash</li>
<li>python38Packages.sip</li>
<li>python38Packages.stytra</li>
<li>python38Packages.uranium</li>
<li>python38Packages.vega</li>
<li>python38Packages.weboob</li>
<li>python38Packages.woob</li>
<li>python39Packages.ansible-kernel</li>
<li>python39Packages.binwalk-full</li>
<li>python39Packages.enamlx</li>
<li>python39Packages.jupyter</li>
<li>python39Packages.lightparam</li>
<li>python39Packages.mayavi</li>
<li>python39Packages.mlt</li>
<li>python39Packages.poppler-qt5</li>
<li>python39Packages.pyqt-builder</li>
<li>python39Packages.pyqt5</li>
<li>python39Packages.pyqt5_with_qtmultimedia</li>
<li>python39Packages.pyqt5_with_qtwebkit</li>
<li>python39Packages.pyqtgraph</li>
<li>python39Packages.pyqtwebengine</li>
<li>python39Packages.pywebview</li>
<li>python39Packages.qimage2ndarray</li>
<li>python39Packages.qreactor</li>
<li>python39Packages.qscintilla-qt5</li>
<li>python39Packages.qtconsole</li>
<li>python39Packages.quamash</li>
<li>python39Packages.sip</li>
<li>python39Packages.uranium</li>
<li>python39Packages.vega</li>
<li>python39Packages.weboob</li>
<li>python39Packages.woob</li>
<li>qarte</li>
<li>qgis</li>
<li>qgis-unwrapped</li>
<li>qnotero</li>
<li>qpaeq</li>
<li>qradiolink</li>
<li>qutebrowser</li>
<li>rapid-photo-downloader</li>
<li>retext</li>
<li>rmview</li>
<li>scudcloud</li>
<li>shotcut</li>
<li>sonic-pi</li>
<li>soundtracker</li>
<li>sumorobot-manager</li>
<li>tambura</li>
<li>tetraproc</li>
<li>tiscamera</li>
<li>tortoisehg</li>
<li>tuxguitar</li>
<li>urh</li>
<li>veusz</li>
<li>vimiv-qt</li>
<li>virtscreen</li>
<li>vorta</li>
<li>webcamoid</li>
<li>webmacs</li>
</ul>
</details>

SuperSandro2000 reviewed (approved):
diff LGTM, build failures are unrelated.

dotlambda commented on pkgs/development/python-modules/pyqt/pyqt5-fix-dbus-mainloop-support.patch:
```diff
@@ -1,27 +1,5 @@
-Date: Sun, 3 Mar 2019 01:13:46 +0100
-Subject: [PATCH] Fix building on Nix
-
-./configure.py tries to find dbus-python header in dbus-1 includedir
-obtained from pkg-config or from --dbus flag. Unfortunately, when supplied,
-it also uses the flag for locating dbus-1 headers. This fails on Nix,
-since every package is installed into its own immutable tree so we cannot
-use a single directory for both dbus-python and dbus-1. We can fix this by
-using pkg-config for finding dbus-python headers too.
-
-Additionally, the build system also tries to install the dbus support module
-to dbus-python tree. Often, it is possible to handle this in pkgconfig as well [1]
-but unfortunately, dbus-python does not export the moduledir in its pc file
-so I have decided to solve this with an extra configure flag.
-
-[1]: https://www.bassi.io/articles/2018/03/15/pkg-config-and-paths/
```
It would be nice to preserve this explanation.

eduardosm replied:
Done

dotlambda commented on pkgs/development/python-modules/pyqtwebengine/default.nix:
```diff
@@ -4,25 +4,23 @@
-  format = "other";
+  format = "pyproject";
+
+  disabled = !isPy3k;
```
```suggestion
  disabled = isPy27;
```

eduardosm replied:
Changed

dotlambda commented on pkgs/development/python-modules/pyqt/5.x.nix:
```diff
@@ -47,19 +33,21 @@ let
+  version = "5.15.4";
+  format = "pyproject";
+
+  disabled = !isPy3k;
```
```suggestion
  disabled = isPy27;
```

eduardosm replied:
Changed

dotlambda commented on pkgs/development/python-modules/pyqtwebengine/default.nix:
```diff
@@ -42,49 +41,19 @@ in buildPythonPackage rec {
-      --sipdir="$out/share/sip/PyQt5" \
-      --pyqt-sipdir="${pyqt5}/share/sip/PyQt5" \
-      --stubsdir="$out/${python.sitePackages}/PyQt5"
+  dontConfigure = true;
```
Is this needed?

eduardosm replied:
It is, otherwise it will try to run qmake

dotlambda replied:
Can you add a comment stating that?

eduardosm replied:
Done

laikq commented on pkgs/applications/graphics/veusz/default.nix:
```diff
@@ -25,12 +25,12 @@ python3Packages.buildPythonApplication rec {
   '';
 
-  # For some reason, if sip5 is found on the PATH, the option --sip-dir is
+  # For some reason, if sip6 is found on the PATH, the option --sip-dir is
```
```suggestion
  # because the sip path does not end in PyQt5, we cannot use the --sip-dir option to set it
  # see https://github.com/veusz/veusz/blob/53b99dffa999f2bc41fdc5335d7797ae857c761f/pyqtdistutils.py#L298
```

The comment is referencing sip5, not sip6, so if we change to sip4 it will no longer apply.

We still need the patch, but for other reasons: The sip dir does not end in PyQt5.

laikq commented on pkgs/applications/graphics/veusz/default.nix:
```diff
@@ -39,7 +39,7 @@ python3Packages.buildPythonApplication rec {
     # --sip-dir does nothing here, but it should be the correct way to set the
     # sip_dir, so I'm leaving it here for future versions
-    "--sip-dir=${python3Packages.pyqt5}/share/sip"
+    "--sip-dir=${python3Packages.pyqt5}/${python3Packages.python.sitePackages}/PyQt5/bindings"
```
```suggestion
```

Previously, I used the `substituteInPlace` because the build script ignored the `--sip-dir` flag. But now, the `--sip-dir` flag is not ignored, but we still need to override it because it does not end in `PyQt5`. I would be in favor of removing this flag, since the build works without it.

eduardosm replied:
I had quickly fixed the build without paying much attention to some details. I updated the comment explaining the new issues that came after building pyqt5 with sip 6.

mikepurvis commented:
Should this change had retained a `sip_5`?

dotlambda commented:
> Should this change have retained a `sip_5`?

No, versioned attributes withing `pythonPackages` are not allowed.

mikepurvis commented:
@dotlambda Isn't there still a `sip_4`? Is that one being grandfathered in?

I have a package (in ROS) which requires patching to use with SIP 4, but is not yet ready for SIP 6. SIP 5 support (for Arch) was added, so that would be the sweet spot for me.

The package: https://github.com/ros-visualization/python_qt_binding/blob/main/cmake/sip_configure.py

The patching: https://github.com/lopsided98/nix-ros-overlay/blob/54fb4baeb206eead92ba68a9a593c3607c1c4f8c/distros/distro-overlay.nix#L230-L250
